### PR TITLE
Milestone 1: Functional Todo-List

### PR DESCRIPTION
**Pull request by sneeu-leeu**
### In this Milestone I :
- Created a JS, HTML and CSS webapp all compiled with Webpack
- Followed Instructions
- Hopefully succeeded

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+## Acknowledgments
+
+- Hat tip to anyone whose code was used
+- Inspiration
+- etc
+
+## 📝 License
+
+This project is [MIT](./MIT.md) licensed.
```
Comment (MahmoudBakr23) [inline README.md]: - [ ] For professional repo, kindly fix the broken MIT link 😊🙏🏻

File: src/index.js [hunk 2]
```diff
@@ -1,14 +1,50 @@
...
+function renderTodo(todo) {
+  const list = document.querySelector('.js-todo-list');
 
-  return element;
+  const isChecked = todo.checked ? 'done' : '';
+  const node = document.createElement('li');
+  node.setAttribute('class', `todo-item ${isChecked} border flex height pad-left`);
+  node.setAttribute('data-key', todo.index);
+  node.innerHTML = `
+    <div class="left flex">
```
Comment (MahmoudBakr23) [inline src/index.js:26]: - [ ] As described in the project requirements, please manage some default tasks to be rendered and sorted on page load by the index number, and not the Time. Please see the next comment about that matter 🙏🏻😊

File: src/index.js [hunk 3]
```diff
@@ -1,14 +1,50 @@
...
+  list.append(node);
 }
 
-document.body.appendChild(component());
\ No newline at end of file
+function addTodo(text) {
+  const todo = {
+    text,
+    checked: false,
+    index: Date.now(),
```
Comment (MahmoudBakr23) [inline src/index.js]: - [ ] It is good thinking to sort the array by the time, but according to the project requirements and the requirements of the next milestones of the same app, you'll have to sort the array by index number. For example, `index: 0`

**Review by MahmoudBakr23 — CHANGES_REQUESTED** [review]
Hello @sneeu-leeu 😃, 

Great work and effort you put into this project so far.

## Highlights 🚀 

There are some excellent points that you have made perfectly. Keep up! 💪
- Linter is correct
- Webpack is well-configured
- Descriptive README, PR headline, and summary
- Live demo link

However, there is always room for improvement.

> Keep in mind that your long hours of work are appreciated. However, the code review process is for improving the code you wrote, and not for criticizing it. 😄


 ## Required Changes ♻️
_Please check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

N/A

Cheers and Happy coding!👏👨‍💻💻

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/index.js (hunk 2 above)
Comment (Davidosky007) [reply src/index.js:26]: * done 👍

Comment (Davidosky007) [reply src/index.js:26]: * done 👍

File: src/index.js (hunk 3 above)
Comment (Davidosky007) [reply src/index.js]: * done 👍

**Comment by Davidosky007** [comment]
Hi @sneeu-leeu, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
## Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Pull request merged**